### PR TITLE
#5773: Move SD model to demo folder

### PR DESCRIPTION
**Pull request by Sudharsan-V**
This PR has the following:

- Moved SD to `models/demos/wormhole/` folder.
- Added the SD tests to run_wh_b0_only.sh

### CODE REVIEW

**Review by AleksKnezevic — APPROVED** [review]

**Comment by AleksKnezevic** [comment]
> This PR has the following:
> 
> * Moved SD to `models/demos/grayskull/` folder.
> * Added the SD tests to run_wh_b0_only.sh

This is a type, the model was correctly moved to `models/demos/wormhole/` in PR.

File: tests/scripts/nightly/run_wh_b0_only.sh [hunk 1]
```diff
@@ -12,6 +12,26 @@ echo "Running nightly tests for WH B0 only"
 env pytest tests/ttnn/integration_tests/unet                # -> failing: issue #7556
 SLOW_MATMULS=1 WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml env pytest tests/ttnn/integration_tests/stable_diffusion
 
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_basic_transformer_block.py
```
Comment (tt-rkim) [inline tests/scripts/nightly/run_wh_b0_only.sh:15]: Are there tests in this folder that cannot be run in nightly, ex. perf, device perf etc? Is this why you list so many?

**Review by tt-rkim — COMMENTED** [review]
Please run nightly and post a passing link for your branch

File: tests/scripts/nightly/run_wh_b0_only.sh [hunk 2]
```diff
@@ -12,6 +12,26 @@ echo "Running nightly tests for WH B0 only"
...
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_resnet_block_2d.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_sharded_attention.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_sharded_matmuls.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_transformer_2d_model.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_ttnn_cross_attention_down_block_2d.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_unet_2d_condition_model.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_unet_mid_block_2d_cross_attn.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_upblock_2d.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_upsample_2d.py
+env pytest tests/ttnn/integration_tests/stable_diffusion/test_upsample_nearest_2d.py
```
Comment (AleksKnezevic) [inline tests/scripts/nightly/run_wh_b0_only.sh:33]: These tests will need to be run with env vars:
`SLOW_MATMULS=1 WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml `

**Comment by Sudharsan-V** [comment]
> Please run nightly and post a passing link for your branch

Sure, will paste the link once the tests pass

File: tests/scripts/nightly/run_wh_b0_only.sh (hunk 2 above)
Comment (Sudharsan-V) [reply tests/scripts/nightly/run_wh_b0_only.sh]: @AleksKnezevic, will add these environment variables. Would it be enough to add the test in a single line? like
`SLOW_MATMULS=1 WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml env pytest tests/ttnn/integration_tests/stable_diffusion/tests/`

File: tests/scripts/nightly/run_wh_b0_only.sh (hunk 1 above)
Comment (Sudharsan-V) [reply tests/scripts/nightly/run_wh_b0_only.sh]: Yes, we need to check these tests in the Nightly, that's why I have added all the tests here.

File: tests/scripts/nightly/run_wh_b0_only.sh (hunk 2 above)
Comment (AleksKnezevic) [reply tests/scripts/nightly/run_wh_b0_only.sh]: yeah, that's fine. I believe there's already a `SLOW_MATMULS=1 WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml env pytest tests/ttnn/integration_tests/stable_diffusion` line just above these additions

**Comment by Sudharsan-V** [comment]
> > This PR has the following:
> > 
> > * Moved SD to `models/demos/grayskull/` folder.
> > * Added the SD tests to run_wh_b0_only.sh
> 
> This is a type, the model was correctly moved to `models/demos/wormhole/` in PR.

Thanks for pointing @AleksKnezevic, I have updated the comment.

**Comment by tt-rkim** [comment]
Did you take out the test invocations from the nightly scripts?

**Comment by Sudharsan-V** [comment]
> Did you take out the test invocations from the nightly scripts?

Yes, the tests are removed from the run_wh_b0_only.sh script.

**Comment by tt-rkim** [comment]
Did you confirm that nightly fast dispatch and the perf pipelines can still find their tests?

**Comment by tt-rkim** [comment]
This  seems like a significant regression on WH on-device SD:

```
AssertionError: Some model(s) AVG DEVICE KERNEL SAMPLES/S are too slow, see above for details. {'AVG DEVICE KERNEL SAMPLES/S': [('stable_diffusion_1batch_', 8.2062, 9.7911)]}
```

@AleksKnezevic should we instruct @Sudharsan-V to investigate?

**Comment by saichandax** [comment]
Device perf CI: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/9267454167)

**Comment by Sudharsan-V** [comment]
@tt-rkim, Please find the CIs for this PR below:
[All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/9280618499) - Passed
[Nightly fast dispatch](https://github.com/tenstorrent/tt-metal/actions/runs/9280620261) - Passed
[Device Perf](https://github.com/tenstorrent/tt-metal/actions/runs/9280621896) - Passed
[Model Perf](https://github.com/tenstorrent/tt-metal/actions/runs/9280622972) - Passed

**Review by eyonland — APPROVED** [review]

File: tests/device_perf_tests/stable_diffusion/test_perf_stable_diffusion.py [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    return noise_pred
+
+
+@skip_for_grayskull()
+@pytest.mark.models_performance_bare_metal
+@pytest.mark.parametrize("device_l1_small_size", [32768], indirect=True)
+@pytest.mark.parametrize(
+    "batch_size, num_inference_steps, expected_compile_time, expected_inference_time",
+    [
+        (1, 4, 3600, 2.6),  # Issue 7816 Inference time
```
Comment (AleksKnezevic) [inline tests/device_perf_tests/stable_diffusion/test_perf_stable_diffusion.py:83]: When did you copy the files over? Latest e2e perf is 0.28

File: tests/device_perf_tests/stable_diffusion/test_perf_stable_diffusion.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        comments=comment,
+    )
+    logger.info("Exit SD perf test")
+
+
+@skip_for_grayskull()
+@pytest.mark.models_device_performance_bare_metal
+@pytest.mark.parametrize(
+    "expected_perf",
+    ((8.23),),
```
Comment (AleksKnezevic) [inline tests/device_perf_tests/stable_diffusion/test_perf_stable_diffusion.py:244]: Same here, latest device perf is 10.31

**Review by AleksKnezevic — CHANGES_REQUESTED** [review]
Perf needs to be updated

**Review by arakhmati — APPROVED** [review]

**Comment by Sudharsan-V** [comment]
@AleksKnezevic. Updated the test files with the recent performance numbers. Additionally, in the current pipeline, Stable Diffusion uses `PNDMScheduler`, hence updated test_perf_stable_diffusion.py is updated to use PNDMScheduler.

Please find the CIs for this PR below:
[All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/9302082222) 
[Nightly fast dispatch](https://github.com/tenstorrent/tt-metal/actions/runs/9302083659) 
[Device Perf](https://github.com/tenstorrent/tt-metal/actions/runs/9302085043)
[Model Perf](https://github.com/tenstorrent/tt-metal/actions/runs/9302086295)

**Comment by Sudharsan-V** [comment]
CI links:
[All post commits](https://github.com/tenstorrent/tt-metal/actions/runs/9313026811)
[Nightly fast dispatch](https://github.com/tenstorrent/tt-metal/actions/runs/9313028529)
[Device perf](https://github.com/tenstorrent/tt-metal/actions/runs/9313029831)
[Model perf](https://github.com/tenstorrent/tt-metal/actions/runs/9313031287)

**Review by AleksKnezevic — APPROVED** [review]

**Review by tt-rkim — APPROVED** [review]

**Review by tt-nshanker — APPROVED** [review]

**Review by tapspatel — APPROVED** [review]

**Pull request merged**